interfaces/builtin/iotedge.go

```diff
@@ -0,0 +1,55 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const iotedgeSummary = `allows Azure iotedge daemon and client`
+
+const iotedgeBaseDeclarationSlots = `
+  azure-iotedge:
+    allow-installation: false
+    deny-connection: true
+    deny-auto-connection: true
+`
+
+const iotedgeConnectedPlugAppArmor = `
+# Description: allow access to the iotedge daemon socket and
+# tasks to manage containers. This gives privileged access to the system
+# via iotedge's socket API to iotedged, which then talks to dockerd
+
+# Description: control and tracing of dockerd containers
+@{PROC}/[0-9]*/environ r,
+ptrace (read),
+capability sys_ptrace,
```

Comment:
do we know what these accesses are for? docker tries to do the same access iirc but still works fine without providing this, though there are more specific ptrace rules in docker-support as well

Reply:
I believe lots is for tracking containers.

Reply:
perhaps it's fine then, however I see that kubernetes-support has the same access for `/proc/.../environ`, albeit that is for the specific case of transitioning to systemd-run, so it's not quite clear-cut. Perhaps the `/proc/.../environ` access should be put into `process-control` along with `capability sys_ptrace` and `ptrace (read)`, but `process-control` is not a superprivileged interface so perhaps that's not a good fit either.

Reply:
And the true is that I'm not 100% sure if this is or it is not problem for iotedged. I have not run enough tests to validate, and we have no source. So hard one to say :(